### PR TITLE
Websocket handshake cleanup

### PR DESCRIPTION
**Pull request by asvetlov**
`do_handshake` is very private API, let's move the implementation into `web.WebSocketResponse` -- the only user of the function.

### CODE REVIEW

File: aiohttp/http.py [hunk 1]
```diff
@@ -27,7 +28,7 @@
 
     # .http_websocket
     'WS_CLOSED_MESSAGE', 'WS_CLOSING_MESSAGE', 'WS_KEY',
-    'WebSocketReader', 'WebSocketWriter', 'do_handshake',
+    'WebSocketReader', 'WebSocketWriter', 'ws_ext_gen', 'ws_ext_parse',
```
Comment (kxepal) [inline aiohttp/http.py:31]: Why expose `ws_ext_gen` and `ws_ext_parse` then?

Comment (asvetlov) [reply aiohttp/http.py:31]: It's a part of internal API, why not.

My main intention for `do_handshake` changes is not preventing exposing new functions but not converting `http` exceptions into `web` ones.
PR raises proper exceptions directly. I believe it simplifies understanding a little, plus stack traces are shorter.

Comment (kxepal) [reply aiohttp/http.py:31]: Aha, I see. Thanks!

**Comment by kxepal** [comment]
Oh, is that [reference](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_ws.py#L98) is about just dropped `do_handshake()`?

**Comment by asvetlov** [comment]
The code is for do nothing on second `prepare` call.
Not related to `do_handshake()` details.

**Comment by kxepal** [comment]
Ok, good then.

**Pull request merged**